[Workers] Added how to filter 504s (#8013)
cloudflare/cloudflare-docs: content/workers/runtime-apis/cache.md

#### Errors

`cache.match` returns a `504` error when the content is stale. 

> You may see an influx of `504` responses with the `RequestSource` of `earlyHintsCache` in Cloudflare Logs when Early Hints is enabled, which is expected and benign. Requests from `earlyHintsCache` are internal subrequests for cached Early Hints, and they are neither end user requests, nor do they go to your origin. Their response status only indicates whether there are cached Early Hints for the request URI (`200` on cache hit, `504` on cache miss). These requests are already filtered out in other views, such as Cache Analytics. To filter out these requests or to filter requests by end users of your website only, refer to [Filter end users](/analytics/graphql-api/features/filtering/#filter-end-users).

Reviewer: ```suggestion
You may see an influx of `504` responses with the `RequestSource` of `earlyHintsCache` in Cloudflare Logs when [Early Hints](/cache/about/early-hints/) is enabled. This behavior is expected and benign. Requests from `earlyHintsCache` are internal subrequests for cached Early Hints. These requests are not end user requests, nor do they go to your origin server. Their response status only indicates whether there are cached Early Hints for the request URI (`200` on cache hit, `504` on cache miss). These requests are already filtered out in other views, such as Cache Analytics. To filter out these requests or to filter requests by end users of your website only, refer to [Filter end users](/analytics/graphql-api/features/filtering/#filter-end-users).
```

> You may see an influx of `504` responses with the `RequestSource` of `earlyHintsCache` in Cloudflare Logs when [Early Hints](/cache/about/early-hints/) is enabled. This behavior is expected and benign. Requests from `earlyHintsCache` are internal subrequests for cached Early Hints. These requests are not end user requests, nor do they go to your origin server. Their response status only indicates whether there are cached Early Hints for the request URI (`200` on cache hit, `504` on cache miss). These requests are already filtered out in other views, such as Cache Analytics. To filter out these requests or to filter requests by end users of your website only, refer to [Filter end users](/analytics/graphql-api/features/filtering/#filter-end-users).

Reviewer 2: I think we actually want to reference [edgeWorkerCacheAPI](https://developers.cloudflare.com/logs/reference/clientrequestsource/) as the request source here, since this page is talking about the Cache API product, not Early Hints which already has its own disclaimer on its own page.

Author: Thanks @drcaramelsyrup Could you help rephrasing the paragraph?

Reviewer 2: I would like a Workers team opinion as well, but perhaps:

> `cache.match` returns a `504` error when the content is stale or not found in cache.

> If you use Cloudflare Logs, you may see these `504` responses with the `RequestSource` of `edgeWorkerCacheAPI`. Again, these are expected if the cached asset was evicted or stale. Note that `edgeWorkerCacheAPI` requests are already filtered out in other views, such as Cache Analytics. To filter out these requests or to filter requests by end users of your website only, refer to [Filter end users](/analytics/graphql-api/features/filtering/#filter-end-users).

Author: Thanks @drcaramelsyrup 
@deadlypants1973 can we add someone from the Workers team to review?

Reviewer 3: I agree that mentioning Early Hints doesn't make sense here.

I offer a suggestion above on how I would word the first paragraph.

For the second paragraph, I think @drcaramelsyrup's suggestion is good. I think I'd suggest the phrase "missing or expired" instead of "evicted or stale", though. "Evicted" and "stale" both imply that the response was cached in the first place, which may not be the case. And I think "expired" is a bit clearer than "stale" -- we use the latter term in documentation elsewhere to refer to expired responses which we knowingly expose to the eyeball, which isn't happening here.
